### PR TITLE
MAINT: Update pre-commit config to fix build

### LINKED ISSUE #1242: BUG: Pre-commit hook is failing
Pre-commit hook is failing with
```
RuntimeError: The Poetry configuration is invalid:
  - [extras.pipfile_deprecated_finder.2] 'pip-shims<=0.3.4' does not match '^[a-zA-Z-_.0-9]+$'
```

_Originally posted by @mattip in https://github.com/airspeed-velocity/asv/issues/1240#issuecomment-1436000372_

### PR DESCRIPTION
**Pull request by HaoZeke**
Closes #1242.

### CODE REVIEW

**Comment by HaoZeke** [comment]
cc @mattip :)

File: setup.cfg [hunk 1]
```diff
@@ -48,7 +48,7 @@ doc =
     sphinx_bootstrap_theme
 dev =
     flake8
-    isort
+   isort >= 5.11.5
```
Comment (mattip) [inline setup.cfg:51]: Is this line missing a space?

Comment (HaoZeke) [reply setup.cfg]: Yup, sorry, fixed.

**Comment by mattip** [comment]
Thanks @HaoZeke

**Pull request merged**